src/index.ts

```diff
@@ -0,0 +1,58 @@
+// =======================
+// Domain: Serial Event Queue
+//
+// Use this to ensure asynchronous SBP invocations get invoked serially,
+// one after another, waiting for the previous invocation to finish completely.
+// =======================
+
+import type { SbpInvocation } from '@sbp/sbp'
+import sbp from '@sbp/sbp'
+
+type EventQueueEvent = {
+  sbpInvocation: SbpInvocation;
+  promise: Promise<unknown>;
+}
+type ThisType = {
+  eventQueues: Record<string, { events: EventQueueEvent[] }>
+}
+
+export default (sbp('sbp/selectors/register', {
+  'okTurtles.eventQueue/_init': function (this: ThisType) {
+    this.eventQueues = Object.create(null)
+  },
+  'okTurtles.eventQueue/isWaiting': function (this: ThisType, name: string): boolean {
+    return !!this.eventQueues?.[name].events.length
```

Comment:
Should this have a `?.` after `[name]`?

Reply:
You're right, the `?` should follow `[name]` since it's `this.eventQueues[name]` that could be undefined.